src/main/cpp/MSNumpressTest.cpp

```diff
@@ -109,6 +108,87 @@ void decodeLinearNice() {
 	cout << "+ pass    decodeLinearNice " << endl << endl;
 }
 
+void decodeLinearNiceLowFP() {
+  
+  double mzs[7];
+  
+  mzs[0] = 100.0;
+  mzs[1] = 200.0;
+  mzs[2] = 300.00005;
+  mzs[3] = 400.00010;
+  mzs[4] = 450.00010;
+  mzs[5] = 455.00010;
+  mzs[6] = 700.00010;
+  
+  size_t nMzs = 7;
+  unsigned char encoded[28];
+
+  // check for fixed points
+  {
+    double fixedPoint;
+    fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 0.1);
+    assert( abs(5 - fixedPoint) < 0.000005);
+
+    fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 1e-3);
+    assert( abs(500 - fixedPoint) < 0.000005);
+
+    fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 1e-5);
+    assert( abs(50000 - fixedPoint) < 0.000005);
+
+    fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 1e-7);
+    assert( abs(5000000 - fixedPoint) < 0.000005);
+
+    // cannot fulfill accuracy of 1e-8
+    fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 1e-8);
+    assert( abs(-1 - fixedPoint) < 0.000005);
+  }
+
+  {
+    double fixedPoint = ms::numpress::MSNumpress::optimalLinearFixedPointMass(&mzs[0], nMzs, 0.001);
+    size_t encodedBytes = ms::numpress::MSNumpress::encodeLinear(&mzs[0], nMzs, &encoded[0], fixedPoint);
+
+    double decoded[7];
+    size_t numDecoded = ms::numpress::MSNumpress::decodeLinear(&encoded[0], encodedBytes, &decoded[0]);
+    assert(25 == encodedBytes);
+    assert(7 == numDecoded);
+
+    assert(abs(100.0 - decoded[0]) < 0.001);
+    assert(abs(200.0 - decoded[1]) < 0.001);
+    assert(abs(300.00005 - decoded[2]) < 0.001);
+    assert(abs(400.00010 - decoded[3]) < 0.001);
+  }
+  
+  double mz_err[7];
+  double encodedLength[7];
+
+  // for higher accuracy, we get longer encoded lengths
+  mz_err[0] = 0.1;  encodedLength[0] = 22;
+  mz_err[1] = 1e-3; encodedLength[1] = 25;
+  mz_err[2] = 1e-5; encodedLength[2] = 29;
+  mz_err[3] = 1e-6; encodedLength[3] = 30;
+  mz_err[3] = 1e-7; encodedLength[3] = 31;
```

Comment:
Good! I think you have overwrite the `ms_err` and `encodedLength` of L168 though...